game-core/src/main/java/games/strategy/triplea/delegate/battle/steps/retreat/OffensiveGeneralRetreat.java

```diff
@@ -166,6 +169,7 @@ private void retreat(
 
     if (battleState.filterUnits(ALIVE, OFFENSE).isEmpty()) {
```

Comment:
It is starting to look like battleActions should own the stack clear logic. That would guarantee that we have not missed any places and could be less wiring code. Instead of wiring both the battle actions and an execution stack, we'd only need the battle actions. Can battle actions be given a reference to the execution stack? It almost seems like they should be combined into the same object.

Reply:
battle actions is just `MightFightBattle` which does have a reference to an execution stack but I'm not sure if it is "the" execution stack during the actual running.  So, I'd rather keep passing around the execution stack so that it can be save compatible.